tests/benchq/resource_estimation/test_openfermion_re.py

```diff
@@ -123,14 +135,160 @@ def test_monotonicity_of_duration_wrt_SCC_time(SCC_time_low, SCC_time_high):
+    """
+    assert best_toffoli.extra.failure_prob / best_T.extra.failure_prob == 2.0
+
+
+@pytest.mark.parametrize(
+    "n_toffoli,n_T",
+    [
+        (20, 20),
+        (40, 40),
+    ],
+)
+def test_calc_of_Algorithm_Failure_Prob(n_toffoli, n_T):
+    num_logical_qubits = 12
+    SCC_time = datetime.timedelta(microseconds=2)
+    physical_error_rate = 1.0e-4
+    best_toffoli = get_physical_cost(
+        num_logical_qubits=num_logical_qubits,
+        num_toffoli=n_toffoli,
+        surface_code_cycle_time=SCC_time,
+        physical_error_rate=physical_error_rate,
+    )
+    best_T = get_physical_cost(
+        num_logical_qubits=num_logical_qubits,
+        num_T=n_T,
+        surface_code_cycle_time=SCC_time,
+        physical_error_rate=physical_error_rate,
+    )
+    """
+    X+Yf - (X+Yf/2) = Yf/2, where X+Yf and X+Yf/2 are
+    Algorithm Failure Prob for Toffoli and T cases respectively,
+    where X -> data failure, Yf -> distillation
+    failure, Y=#Toffoli gate=#T gates and f is 1 CCZ error.
+    If failure rate is calculated correctly,
+    to ascertain if Algorithm Failure Probability is calculated correctly
+    for both toffoli and T.
+    """
+
+    numpy.testing.assert_almost_equal(
+        2 * (best_toffoli.logical_error_rate - best_T.logical_error_rate) / n_toffoli,
+        best_toffoli.extra.failure_prob,
+    )
+
+    numpy.testing.assert_almost_equal(
+        (best_toffoli.logical_error_rate - best_T.logical_error_rate) / n_T,
+        best_T.extra.failure_prob,
+    )
+
+
+def test_default_T_factories():
+    num_logical_qubits = 12
+    SCC_time = datetime.timedelta(microseconds=2)
+    physical_error_rate = 1.0e-3
+    # Default Magic State Factory details
+    # used based on this physical error rate.
+    num_T = 200
+    num_toffoli = 140
+    best_T = get_physical_cost(
+        num_logical_qubits=num_logical_qubits,
+        num_T=num_T,
+        surface_code_cycle_time=SCC_time,
+        physical_error_rate=physical_error_rate,
+    )
+    best_toffoli = get_physical_cost(
+        num_logical_qubits=num_logical_qubits,
+        num_toffoli=num_toffoli,
+        surface_code_cycle_time=SCC_time,
+        physical_error_rate=physical_error_rate,
+    )
+    assert (
+        best_T.extra.rounds == 186
+        and best_T.extra.failure_prob == 3.6000000000000003e-16
+    )
+    assert (
+        best_toffoli.extra.rounds == 186
+        and best_toffoli.extra.failure_prob == 3.6000000000000003e-16
+    )
+
+
+def test_default_values():
+    num_logical_qubits = 12
+    physical_error_rate = 1.0e-4
+    """
+    If physical_error_rate != default value,
+    this ascertains if both num_T=0 and num_toffoli=0,
+    then it must throw an error.
+    """
+    with pytest.raises(SystemExit) as dvalue:
+        a, b = get_physical_cost(
+            num_logical_qubits=num_logical_qubits,
+            physical_error_rate=physical_error_rate,
+        )
+
+    assert dvalue.type == SystemExit
```

Comment:
This should be updated according to how you implement the error handling from Athena's suggestion here.
I think it might work to just replace the with pytest.raises(SystemExit)... line with the line
with pytest.raises(ValueError):

Reply:
Done!